src/op_analytics/cli/subcommands/pulls/agora/data_access.py

```diff
@@ -0,0 +1,235 @@
+import glob
+import os
+from enum import Enum
+from functools import cache
+import re
+import polars as pl
+import pyarrow as pa
+
+from op_analytics.coreutils.duckdb_inmem import init_client
+from op_analytics.coreutils.env.aware import OPLabsEnvironment, current_environment
+from op_analytics.coreutils.logger import structlog
+from op_analytics.coreutils.partitioned import (
+    DataLocation,
+    DataWriter,
+    ExpectedOutput,
+    OutputData,
+    SinkMarkerPath,
+    SinkOutputRootPath,
+)
+from op_analytics.coreutils.partitioned.breakout import breakout_partitions
+from op_analytics.coreutils.partitioned.dataaccess import DateFilter, MarkerFilter, init_data_access
+from op_analytics.coreutils.path import repo_path
+from op_analytics.coreutils.time import date_fromstr, now_friendly_timestamp
+from op_analytics.coreutils.rangeutils import DateRange
+
+log = structlog.get_logger()
+
+MARKERS_TABLE = "daily_data_markers"
+
+
+class Agora(str, Enum):
+    """Supported agora datasets."""
+
+    DELEGATES = "delegates_v1"
+    DELEGATE_VOTES = "delegate_votes_v1"
+    PROPOSALS = "proposals_v1"
+
+    def read(
+        self,
+        min_date: str | None = None,
+        max_date: str | None = None,
+        date_range_spec: str | None = None,
+    ):
+        """Read agora data."""
+        return load(
+            dataset_name=self.value,
+            datefilter=self._date_filter(min_date, max_date, date_range_spec),
+            location=DataLocation.GCS,
+        )
+
+    def dump_local_copy(
+        self,
+        min_date: str | None = None,
+        max_date: str | None = None,
+        date_range_spec: str | None = None,
+    ):
+        """Dump a copy of the data to a single local parquet file."""
```

Comment:
for all of the methods in the dataaccess module, could you take a look at defillama and use the functions that were lifted over to coreutils for common use? 

Also could you rename data_access.py to dataaccess.py?  that's just for conforming to the name we use in defillama.  For modules that are meant to be imported frequently I try to not use underscores.

Reply:
I observe that we use Pandas and Polars at the same time so far in both Defillama and Agora. 
If there isn't a reason for Polars to exist, should I port dataaccess.py to using only Pandas?
(re. dataaccess, I thought the missing "_" was a typo) 😁

Reply:
I tend to use polars more, but if there's functionality you'd rather implement in pandas then I'm all for it as well.  Right now I don't think the defillama pulls use pandas. Maybe Chuck uses it for analysis of the pulled data?